export/main.ts

```diff
@@ -0,0 +1,89 @@
+import { PhylumApi } from 'phylum';
+import { parse } from "https://deno.land/std@0.156.0/flags/mod.ts";
+import { MultiProgressBar } from "https://deno.land/x/progress@v1.4.4/mod.ts";
+
+
+/**
+ * Fetch the project data from the Phylum API.
+ */
+async function fetchProjectData(projectId: string, group?: string): Promise<any> {
+    try {
+        const url = group ? `groups/${group}/projects/${projectId}` : `data/projects/${projectId}`;
+        const response = await PhylumApi.fetch("v0/", url, {});
+
+        if(!response.ok) {
+            console.error(`Failed to fetch project data, received HTTP error: ${response.status}`);
+        } else {
+            const data = await response.json();
+            return data;
+        }
+    } catch (error) {
+        console.error("There was an issue fetching the project data:", error);
+    }
+}
+
+/** 
+ * Iterate through the API and fetch all known projects. 
+ */
+async function fetchProjects(cursor?: string, hasMore?: bool = true, perPage: int = 100): Promise<any> {
+    if(!hasMore) {
+        return [];
+    }
+
+    const base = `/projects?paginate.limit=${perPage}`;
+    const projectsUrl = cursor ? `${base}&paginate.cursor=${cursor}` : base; 
+    const response = await PhylumApi.fetch("v0/", projectsUrl, {});
+
+    if (!response.ok) {
+        throw new Error(`Failed to fetch projects, HTTP error: ${response.status}`);
+    }
+
+    const ret =  await response.json(); 
+    const nextCursor = ret.values[ret.values.length - 1].id;
+
+    return ret.values.concat(fetchProjects(nextCursor, ret.has_more));
```

Comment:
```suggestion
    if (ret.has_more) {
        return ret.values.concat(fetchProjects(nextCursor));
    }
    return ret.values;
```
This syntax may not be idiomatic since I don't really know JS/TS. The suggestion allows for the removal of the `hasMore` parameter and the following block:

```
    if (!hasMore) {
        return [];
    }
```

If all the suggestions are applied, the function would look like this:

<details><summary>Details (click to expand...)</summary>
<p>

```
async function fetchProjects(cursor?: string): Promise<any> {
    const base = `/projects?paginate.limit=100`;
    const projectsUrl = cursor ? `${base}&paginate.cursor=${cursor}` : base;
    const response = await PhylumApi.fetch("v0/", projectsUrl, {});

    if (!response.ok) {
        throw new Error(`Failed to fetch projects, HTTP error: ${response.status}`);
    }

    const ret = await response.json();
    const nextCursor = ret.values[ret.values.length - 1].id;

    if (ret.has_more) {
        return ret.values.concat(fetchProjects(nextCursor));
    }
    return ret.values;
}
```

</p>
</details>